tests/semantic_analyser.cpp

```diff
@@ -802,6 +803,23 @@ TEST(semantic_analyser, call_stack)
   test("kprobe:f { @x = 3; ustack(perf, @x) }", 1);
 }
 
+TEST(semantic_analyser, call_macaddr)
+{
+  std::string structs = "struct mac { char addr[6]; }; ";
+
+  test(structs + "kprobe:f { macaddr((struct mac*)arg0); }", 0);
```

Comment:
need some tests (and runtime tests) for the other supported cases too, like

macaddr(arg0)
macaddr(*(struct X*)y)
macaddr(somekindofarray)

Reply:
I'm not sure what you meant for these two:

>macaddr((struct X)y)

How would it differ from this line? Do you mean a cast to a non-pointer struct type?

>macaddr(somekindofarray)

The line below this one is testing an array, no?

Reply:
oh woops, codeblock makes the difference

```
macaddr(arg0)
macaddr(*(struct mac*)y)
macaddr(somekindofarray)
```

> > macaddr(somekindofarray)
> 
> The line below this one is testing an array, no?

I'm not sure, I just want to make sure it works for data that has already been loaded onto the stack (so no probe_read required).